### PR TITLE
[YOLOV8] Update version

### PR DESCRIPTION
**Pull request by dsikka**
### Summary
- Updated deepsparse support for yolov8
- Small similar changes to the ones made for sparseml's yolov8 update
- `process_engine_outputs` has a slight change. Previously providing `nm` for number of masks, which is no longer used by the yolov8 `non_max_suppression` function.

### Testing:

- Tested locally using both sparsezoo and yolov8 models for both segmentation and detection
- Workflows tested: pipeline creation and forward pass, annotation, eval and benchmarking 
 
### sparsezoo models tested:

1. detection: `"zoo:cv/detection/yolov8-m/pytorch/ultralytics/coco/base_quant-none"`
2. segmentation: `"zoo:cv/segmentation/yolov8-n/pytorch/ultralytics/coco/base-none"`

### yolov8 models tested:

1. detection: `yolov8n.pt`
2. segmentation: `yolov8n-seg.pt`

### Pipeline creation + forward pass:

Detection
```
from deepsparse import Pipeline
model_path = "yolov8n.onnx" 
images = ["basilica.jpg"]
yolo_pipeline = Pipeline.create(
    task="yolov8",
    model_path=model_path,
)
pipeline_outputs = yolo_pipeline(images=images)
```

Segmentation:

```
from deepsparse import Pipeline
model_path = "yolov8n-seg.onnx" 
images = ["basilica.jpg"]
yolo_pipeline = Pipeline.create(
    task="yolov8",
    subtask="segmentation",
    model_path=model_path,
)

pipeline_outputs = yolo_pipeline(images=images)
```
### Annotation:

Detection:
`deepsparse.yolov8.annotate --source basilica.jpg --model_filepath "yolov8n.onnx"`
Segmentation:
`deepsparse.yolov8.annotate --source basilica.jpg --model_filepath "yolov8n-seg.onnx" --subtask segmentation`

Results verified for outputs produced by both workflows above

## Eval

Detection:
`deepsparse.yolov8.eval --model_path yolov8n.onnx`
Segementation:
`deepsparse.yolov8.eval --model_path yolov8n-seg.onnx --subtask segmentation`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def _parse_requirements_file(file_path):
     "opencv-python<=4.6.0.66",
 ]
 _yolo_integration_deps = [
-    "torchvision>=0.3.0,<0.14",
+    "torchvision>=0.3.0,<=0.15.1",
```
Comment (abhinavnmagic) [inline setup.py:136]: We just need to verify torchvision==0.15.1 doesn't break other computer vision flows, like IC. Someone might have already tried it.

Comment (dsikka) [reply setup.py:136]: IC?

Comment (dsikka) [reply setup.py:136]: we would have to verify all IC workflows if we want to bump up `_image_classification_deps` to torchvision==0.15.1 there as well

Comment (bfineran) [reply setup.py:136]: let's keep it as separate to this update. given this is just for the yolo install pathway, we only need to guarantee it works for yolo. if someone has both yolo and ic installed then the dependency resolver should take the lower of the max versions

**Review by bfineran — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Review by abhinavnmagic — APPROVED** [review]

**Pull request merged**